### ISSUE TITLE
Poor accessibility of comboboxes

### ISSUE THREAD

**Issue by steverep** [home-assistant/frontend#12952]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

The `ha-combobox` component uses [`vaadin-combo-box-light`]() and inserts a subclass of `mwc-textfield` for the input.  The main problem is that `vaadin-combo-box-light` is setting the ARIA role and other attributes on the custom element, but they need to be set on the `<input>` element in the shadow DOM instead.

However, even if that were achieved, it's still not clear whether the end result would be accessible.  I can only verify that the non-light version is accessible.

In addition, the toggle and clear elements I believe should be buttons.  Currently they are just `ha-svg-icon` elements which are hidden from the accessibility tree, and the `aria-label` has no effect because it's not labeling anything with a implicit or explicit ARIA role.

### Describe the behavior you expected

Final markup should follow the [combobox pattern in the ARIA Authoring Practices Guide](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/).

### Steps to reproduce the issue

1. Download NVDA, use VoiceOver, etc.
2. Go ahead and close your eyes and try to figure out a combobox with keyboard only

### What version of Home Assistant Core has the issue?

2022.6.5

### In which browser are you experiencing the issue with?

Firefox, Chrome, or Safari (latest stable)

### Which operating system are you using to run this browser?

Windows 10 and iOS 15

**Comment by steverep** [comment]
It's also a problem using `mwc-list-item` because it automatically sets `tabindex="-1"` on itself (even when non-interactive!).  So you end up with a vaadin container for each listbox item that is focusable and has the correct ARIA `role="option"` and other attributes, but then user agents will get thoroughly confused with a focusable descendant.

**Comment by steverep** [comment]
I checked the API for `vaadin-combo-box-light` and see no way to fix without a patch from them.  See vaadin/web-components#4050.

**Comment by steverep** [comment]
@bramkragten can you tag this one and others with the accessibility label, or just give me triage permission and I'll take care of it?  It's getting hard to keep them all straight since only some are labelled and I'd like to file more.
